Makefile

```diff
@@ -4,8 +4,8 @@ export CLANG := clang
 export OBJCOPY := llvm-objcopy
 
 CARGO := cargo
-RELEASE_VERSION ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name == "retis") | .version')
-RELEASE_NAME ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq '.packages | .[] | select(.name=="retis") | .metadata.misc.release_name')
+RELEASE_VERSION = $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name == "retis") | .version')
```

Comment:
any specific reason behind this?
The idea here was to allow to create quick scratch builds without editing any file

Reply:
The idea is to make the version predictable so we can trust it (while using retis, in bug reports, in discussions with users, etc). Allowing to set it to a different value makes sense for a custom build but with this PR such builds will have a specific version that we can use to distinguish a proper build (and the commit id should be included which also can help later on to be sure exactly what was built).

IMO this does not prevent to have a custom version, and even improves the situation as we have a better idea what was the tree like when building it.

Do you see other use cases not covered after this PR?

Reply:
no, the PR logic is fine and it is a very nice improvement.
What I was trying to suggest/keep, given you can always change those variables by editing the Makefile (when built from sources :) was the ability to specify a custom name/version (probably the name makes more sense). `?=` doesn't do anything other than easily override the behavior introduced by this PR (something like: "scratch-build-with-this-nice-feature").
This is because -dirty doesn't tell the whole story.

Maybe a "LOCALVERSION" suffix (instead of dirty, if exists) could further improve it? WDYT?

Reply:
> What I was trying to suggest/keep, given you can always change those variables by editing the Makefile (when built from sources :)

Right :) Let's say allowing to override is a built-in feature.

> was the ability to specify a custom name/version (probably the name makes more sense). `?=` doesn't do anything other than easily override the behavior introduced by this PR (something like: "scratch-build-with-this-nice-feature"). This is because -dirty doesn't tell the whole story.
> 
> Maybe (instead of the `?=` thing) a "LOCALVERSION" suffix (as a replacement for dirty, if both exist) could further improve it? WDYT?

Good point, makes sense to allow overriding the version name to have a human readable string. We can keep the version built-in though, to avoid confusion when overriding the version name.

With that, I would keep the overriding of the name built-in and not introduce a localversion to keep things simple (for a feature which is not used too much).

Does that work?